frontend/src/_helpers/handleAppAccess.js

```diff
@@ -26,9 +26,17 @@ export const handleAppAccess = (componentType, slug, version_id) => {
     });
   } else {
     /* Released app link [launch/sharable link] */
-    return appsService.validateReleasedApp(slug).catch((error) => {
-      handleError(componentType, error, redirectPath);
-    });
+    try {
+      return await appsService.validateReleasedApp(slug);
+    } catch (error) {
+      let editPermission = true;
+      try {
+        await appsService.validatePrivateApp(slug, queryParams);
+      } catch (err) {
+        editPermission = false;
+      }
+      handleError(componentType, error, redirectPath, editPermission);
+    }
```

Comment:
Instead of doing two APIs to find if the user has edit access to the app or not. we can add an edit access check from the API level and pass some flag or data with the `validateReleasedApp` API error object. 
for reference: https://github.com/ToolJet/tj-ee/blob/f6d3df49042aa7e34b0e23159604856b59198be1/server/src/controllers/apps.controller.ts#L107